pkg/controller/utils/utils.go

```diff
@@ -438,32 +439,33 @@ func WaitToAddResourceWatch(controller controller.Controller, client kubernetes.
 				duration = maxDuration
 			}
 			ticker.Reset(duration)
-			if isResourceReady(client, obj.GetObjectKind().GroupVersionKind().Kind) {
-				err := controller.Watch(&source.Kind{Type: obj}, &handler.EnqueueRequestForObject{})
-				if err != nil {
-					log.Info("failed to watch %s resource: %v. Will retry to add watch", obj.GetObjectKind().GroupVersionKind().Kind, err)
-				} else {
-					flag.MarkAsReady()
-					return
-				}
+			if ok, err := isResourceReady(client, obj.GetObjectKind().GroupVersionKind().Kind); err != nil {
+				log.Info("failed to list registered groups and resources - will retry: %v", err)
+			} else if !ok {
+				log.Info("waiting for %s resource to be registered - will retry", obj.GetObjectKind().GroupVersionKind().Kind)
+			} else if err := controller.Watch(&source.Kind{Type: obj}, &handler.EnqueueRequestForObject{}); err != nil {
+				log.Info("failed to watch %s resource - will retry: %v", obj.GetObjectKind().GroupVersionKind().Kind, err)
+			} else {
+				flag.MarkAsReady()
```

Comment:
Do we want a success log? Maybe at a lower verbosity?

Reply:
Sure - I think the same verbosity as the log on 434.... But perhaps they could both be downgraded to debug.